### PR TITLE
fix: wrong baseid in QuickImport

### PR DESCRIPTION
**Pull request by semmjon**
## Change Summary

PR for:
- closes https://github.com/nocodb/nocodb/issues/5103

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ x ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

1. The useProject.ts file in the nc-gui package:
A filter has been added to the bases computed property to only include those that have the enabled property set to true.

2. The index.vue file in the nc-gui package:
The baseId property has been added to the openQuickImportDialog function, and it is set to the id of the first base in the bases array that has the enabled property set to true.
In summary, these changes filter out disabled bases in the bases computed property, and pass the id of the first enabled base to the openQuickImportDialog function as the baseId property.

You can test it like in the issue description

### CODE REVIEW

File: packages/nc-gui/composables/useProject.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export const useProject = createSharedComposable(() => {
 
   const project = ref<ProjectType>({})
 
-  const bases = computed<BaseType[]>(() => project.value?.bases || [])
+  const bases = computed<BaseType[]>(() => project.value?.bases?.filter((base: BaseType) => base.enabled) || [])
```
Comment (wingkwong) [inline packages/nc-gui/composables/useProject.ts:38]: By adding this change, it will break the share form views. You can reproduce the issue by creating a form view and then share it. The shared form view will be not shown and you can notice the errors from console.

The reason is that we haven't passed `enabled` in `useSharedFormViewStore`, hence `bases` would be empty (with your new change) causing undefined `sqlUis`.

https://github.com/nocodb/nocodb/blob/ea24fa91211f2bb1c6512504f5289b93b51833ee/packages/nc-gui/composables/useSharedFormViewStore.ts#L90-L98

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: packages/nc-gui/composables/useProject.ts (hunk 1 above)
Comment (semmjon) [reply packages/nc-gui/composables/useProject.ts:38]: I figured that by requiring at least the main database to be enabled (if no others exist/ enabled) that wouldn't happen.

File: packages/nc-gui/pages/[projectType]/[projectId]/index/index/index.vue [hunk 2]
```diff
@@ -98,6 +98,7 @@ function openQuickImportDialog(type: QuickImportTypes, file: File) {
     'modelValue': isOpen,
     'importType': type,
     'onUpdate:modelValue': closeDialog,
+    'baseId': bases.value?.filter((base: any) => base.enabled)[0].id,
```
Comment (wingkwong) [inline packages/nc-gui/pages/[projectType]/[projectId]/index/index/index.vue:101]: don't use `any` when you know the type. Use `BaseType` instead.

**Review by wingkwong — CHANGES_REQUESTED** [review]
Same change should be applied here as well.

https://github.com/nocodb/nocodb/blob/ef72406cb64ef90ff55a84402cca8e2fb6233960/packages/nc-gui/pages/%5BprojectType%5D/%5BprojectId%5D/index/index/index.vue#L129-L133

This is triggered by clicking `Add New Table`.

image

**Review by wingkwong — CHANGES_REQUESTED** [review]
Still not cater 

image

**Review by wingkwong — APPROVED** [review]
